source/ceylon/net/http/server/MatcherRule.ceylon

```diff
@@ -16,6 +17,22 @@ shared abstract class Matcher() {
             => Or(this, other);
 }
 
+"Matcher to leverage Undertow's template mechanism for path templates and path parameters.
+ It should be given to an Endpoint without combining it with other matchers.
+ 
```

Comment:
Does you current solution work with `OR` and fail only when `AND` is used? If that is the case, then we can go this way and maybe deprecate `AND` or leave it as open issue and document it.
If neither `AND` nor `OR` is working then my vote is to fix it before we merge it in.